### ISSUE TITLE
Add KFocusTrap and use it in KModal

### ISSUE THREAD

**Issue by MisRob** [learningequality/kolibri-design-system#746]
🌱 **Are you new to the codebase? Welcome! Please see [the contributing guidelines](https://github.com/learningequality/kolibri-design-system/blob/develop/CONTRIBUTING.md).**

### Blocks

- https://github.com/learningequality/kolibri/issues/12588

### Summary

This issue has two related goals

- Move focus trapping logic from Kolibri to this repository and expose it as a new `KFocusTrap` component
- Use `KFocusTrap` within `KModal` to ensure that users' focus remains contained within the modal interface, preventing accidental interactions with elements outside the modal.

### Guidance

**(1) Create `KFocusTrap`**

Copy [`FocusTrap` from Kolibri](https://github.com/learningequality/kolibri/blob/bbbd09e3f8862968dba2766091edc270508b67d0/kolibri/core/assets/src/views/FocusTrap.vue) to this repository and name it as `KFocusTrap`. Follow the guidance for [creating a new component](https://github.com/learningequality/kolibri-design-system/blob/develop/dev_docs/03_how_to_update_library.md#addupdate-a-component).

**(2) Wrap the content of `KModal` in `KFocusTrap`**

```jsx
// KModal.vue
<template>
  <KFocusTrap>
    <component :is="wrapper" ...>
    ...
```

**(3) Check that the focus trapping works as expected**

on the [the playground page](https://github.com/learningequality/kolibri-design-system/blob/develop/dev_docs/03_how_to_update_library.md#1-preview-updates-in-the-kolibri-design-system-environment).

**(4) Add a new documentation page for `KFocusTrap`**

- Follow ["Create a documentation page for a new component"](https://github.com/learningequality/kolibri-design-system/blob/develop/dev_docs/04_how_to_update_docs.md#create-a-documentation-page-for-a-new-component)
- Follow [“Writing” guidelines](https://design-system.learningequality.org/writing/). Use language that is consistent with other documentation pages.
- Describe briefly the focus trapping function
- A live example loosely similar to the ‘default behavior’ on [https://focus-trap.github.io/focus-trap/](https://focus-trap.github.io/focus-trap/) with code sample would be nice. See `DocsShow` and `DocsShowCode` usage on other documentation pages.

### The Value Add

Improves accessibility of modals.

### Acceptance criteria

- [ ]  There is `KFocusTrap` component including a new documentation page
- [ ]  `KFocusTrap` is used within `KModal` and functions as expected
- [ ]  There are no regressions in `KModal`

**Comment by lokesh-sagi125** [comment]
hey @MisRob  can you assign this to me?

**Comment by MisRob** [comment]
Hey @lokesh-sagi125, assigning, thank you for volunteering

**Comment by lokesh-sagi125** [comment]
hey @MisRob  is there any chat server for LE like discord or is it just github?

**Comment by lokesh-sagi125** [comment]
hey @akolson  @MisRob i keep getting this particular linting issue even though i added the blank lines. ,couldn't find the solution in the documentation , what should i do to fix this?.
<img width="1226" alt="Screenshot 2024-09-02 at 10 13 03 PM" src="https://github.com/user-attachments/assets/eefa2b31-9d9a-4ccd-8582-437e674987ff">

**Comment by akolson** [comment]
Its most likely the line spacing before the <script> tag -- they must be 2 lines. Also the script tag should be at the start of the line and no spaces before it. For example, 

```vue
<template>
    ...
</template>


<script>
    ...
</script>
```

**Comment by MisRob** [comment]
> hey @MisRob is there any chat server for LE like discord or is it just github?

@lokesh-sagi125 Yes, we have a Slack channel for the community of contributors. You're welcome to send an e-mail to contributors@learningequality.org to receive an invitation.

**Comment by iamshobhraj** [comment]
@MisRob Can you please assign this issue to me.

**Comment by AlexVelezLl** [comment]
Hey @iamshobhraj! Thanks for your interest in contributing to this issue! Unfortunately, this is already assigned. But you’re welcome to find [a "help wanted" issue with no assignee](https://github.com/search?q=org%3Alearningequality+is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+no%3Aassignee&type=issues) :).

**Comment by AlexVelezLl** [comment]
Closed by #799
